This improves the experimental feature template

sdake opened the pull request:
Reduce initial work for submission of an experimental feature
- Require an identified lead
- Require a short description of the feature

As RFC/design/documentation work is completed, check off and link each work product.

brian-avery commented on .github/ISSUE_TEMPLATE/experimental_feature.md:
```diff
@@ -1,31 +1,47 @@
+- [ ] Instructions for enabling and using the feature have been documented. This may
+        include instructions for building, running an `istioctl experimental command`, or
 	using the preview profile. 
+- [ ] Feedback plan authoed. This may include user feedback meetings,
```
authored

sdake replied:
clarified.

brian-avery commented on .github/ISSUE_TEMPLATE/experimental_feature.md:
```diff
@@ -1,31 +1,47 @@
+experimental feature. Please complete this template during development.
 
-**Feature Name:** 
+This template shall be completed before users are exposed to any new experimetnal feature.
```
experimental

sdake replied:
resolved.

brian-avery commented on .github/ISSUE_TEMPLATE/experimental_feature.md:
```diff
@@ -1,31 +1,47 @@
-- [ ] Feature is reviewed in a work group meeting
-- [ ] RFC has been written
+- [ ] Review of user stories in a work group meeting
+- [ ] RFC Authored
```
Discussed this in Slack, but we have templates for design docs and RFCs in the Istio drive. We should link to them in the templates.

sdake replied:
agreed applied.

brian-avery commented on .github/ISSUE_TEMPLATE/experimental_feature.md:
```diff
@@ -1,31 +1,47 @@
+
+**Feedback plan:**
 
 > Link to instructions 
```
These should probably be moved to their respective sections or removed

sdake replied:
agreed - good suggestion. applied.

howardjohn commented on .github/ISSUE_TEMPLATE/experimental_feature.md:
```diff
@@ -1,31 +1,47 @@
+
+**Feature name:**
+
+**Primary lead:**
```
```suggestion
**Primary lead(s):**
```

may be more inclusive, many have multiple leads

sdake replied:
agreed, added.

sdake commented:
rendering: https://github.com/sdake/enhancements/blob/master/.github/ISSUE_TEMPLATE/experimental_feature.md

sdake commented:
Seeing the rendering, I believe there are more improvements that can be made to increase brevity (which can improve velocity).

Also, I am not sure how I feel about the links being directly integrated into the checkboxes. This will require individuals to edit the templates with the checkbox links, which may end in doom for some :) @brian-avery any thought on this (this integration was your request).

Finally, and perhaps this is the biggest improvement to be made to all these templates, is adding comments for each field of the template, such that the person filling out the template knows what to do with the template. This is (one) way to improve the brevity of the template.

brian-avery commented:
Yeah, I kind of liked them separate. My comment was more along the lines that we had:

[  ]  Steps for enabling the feature have been documented. This may include instructions for building, running an istioctl experimental command, or using the preview profile.
[  ]  Plan exists for getting feedback. This may include user feedback meetings, discuss.istio.io conversations, GitHub issues, or mailing lists.

>    Link to instructions
>   Plan for feedback

And I was just proposing keeping the boxes that we had for instructions and feedback with the check boxes for the respective content.

brian-avery commented:
By the way, that same rendering can be seen directly from the PR.

sdake commented:
@brian-avery I don't believe I have changed the sprit nor intent of the original template. I did add a few fields which I felt were required and dropped the comments into the document such that they don't render.

Cheers,
-steve

sdake commented:
> Yeah, I kind of liked them separate. My comment was more along the lines that we had:
> 
> [ ] Steps for enabling the feature have been documented. This may include instructions for building, running an istioctl experimental command, or using the preview profile.
> [ ] Plan exists for getting feedback. This may include user feedback meetings, discuss.istio.io conversations, GitHub issues, or mailing lists.
> 
> > Link to instructions
> 
> > Plan for feedback
> 
> And I was just proposing keeping the boxes that we had for instructions and feedback with the check boxes for the respective content.

@brian-avery - ok - makes sense. I did take a stab at documenting the template for the development community, as they feel like the primary audience. The comments (should) be hidden during rendering, but not during form fillout or editng. Although a merge would be required to experiment 👍

Cheers,
-steve

brian-avery commented on .github/ISSUE_TEMPLATE/experimental_feature.md:
```diff
@@ -1,31 +1,64 @@
-This page lists the requirements for experimental features. Experimental features allow Istio work groups to collect more information
-on potential new functionality. This template should be completed before users try out any new experimental features. Please check off and 
-document the steps as they are completed.
+[//]: # (This page lists the requirements for an experimental feature. Completion of this template enables Istio work groups)
```
Cool. I wasn't aware of these.

brian-avery commented on .github/ISSUE_TEMPLATE/experimental_feature.md:
```diff
@@ -1,31 +1,64 @@
+
+- [ ] [Feedback plan](insert_your_link_here).
+
+[//]: # (This may include user feedback meetings, discuss.istio.io conversations, GitHub issues, or mailing lists.)
```
Thanks for adding all of this documentation. This is awesome.

brian-avery commented on .github/ISSUE_TEMPLATE/experimental_feature.md:
```diff
@@ -1,31 +1,64 @@
+[//]: # (An RFC is mandatory to graduate to experimental. The RFC does not have to be reviewed in a work group)
+[//]: # (meeting to graduate to experimental.)
+
+- [ ] [Documention](insert_your_link_here) for enabling and using the feature.
```
Documentation

sdake replied:
resolved.

linsun commented:
I think the newly addition to the template could add a lot more work from developers and many info are already in RFC.  I'll let others to chime in.  Also, if we require this level of details in the experimental template, we need to update alpha/beta/stable template as well.

brian-avery commented:
@linsun I don't think anything was actually added. @sdake rephrased the content and added comments explaining what each field was asking for.

sdake commented:
New feature request can be viewed here: https://github.com/sdake/enhancements/issues/new?template=experimental_feature.md

sdake commented:
@brian-avery to be fair, I did add two fields:
1. the leads field
1. the short description field

everything else is just reword / reorganization / comments.

sdake commented:
rendering of the page on github: https://github.com/sdake/enhancements/blob/master/.github/ISSUE_TEMPLATE/experimental_feature.md

linsun commented:
sorry - my bad @sdake @brian-avery , i had thought it is adding a lot of new things initially but realized it is more adding clarifications.  Yes, updating 4 templates would be nice.

linsun commented:
> rendering of the page on github: https://github.com/sdake/enhancements/blob/master/.github/ISSUE_TEMPLATE/experimental_feature.md

rendering LG!  hard to tell from just reviewing the lines :)

sdake commented:
@linsun would you mind approving this - I will follow up with comments for the others after the 20th. I want to submit an experimental feature for the ingress gateway changes.

Thanks,
-steve
